### ISSUE TITLE
Upload fails

### ISSUE THREAD

**Issue by pvagner** [haiwen/seadroid#841]
I'm running Seafile v 2.2.20 installed from f-droid.

Uploading a file does not work at all.

I have used it successfully approx two months ago.

I'll try to collect more details.

Besides access to storage, seafile does not ask for more permissions for me.
I am using Android 9 october 2019 security update on Nokia 6 TA-1021.

This is the crash I'm getting:

```
--------- beginning of crash
11-14 08:40:19.286 29200 29200 E AndroidRuntime: FATAL EXCEPTION: main
11-14 08:40:19.286 29200 29200 E AndroidRuntime: Process: com.seafile.seadroid2, PID: 29200
11-14 08:40:19.286 29200 29200 E AndroidRuntime: java.lang.SecurityException: Permission Denial: startForeground from pid=29200, uid=10161 requires android.permission.FOREGROUND_SERVICE
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:1950)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:1918)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:1868)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:5142)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:695)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.seafile.seadroid2.notification.UploadNotificationProvider.notifyStarted(UploadNotificationProvider.java:82)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.seafile.seadroid2.notification.BaseNotificationProvider.updateNotification(BaseNotificationProvider.java:78)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.seafile.seadroid2.transfer.UploadTaskManager.notifyProgress(UploadTaskManager.java:72)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.seafile.seadroid2.transfer.UploadTaskManager.onFileUploadProgress(UploadTaskManager.java:103)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.seafile.seadroid2.transfer.UploadTask.onProgressUpdate(UploadTask.java:68)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.seafile.seadroid2.transfer.UploadTask.onProgressUpdate(UploadTask.java:18)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:715)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:198)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6729)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.enforcePermission(ActivityManagerService.java:9803)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:1238)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:889)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:21319)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 	at android.app.IActivityManager$Stub.onTransact$setServiceForeground$(IActivityManager.java:10588)
11-14 08:40:19.286 29200 29200 E AndroidRuntime: 
11-14 08:40:19.298 29200 29200 I Process : Sending signal. PID: 29200 SIG: 9
11-14 08:40:19.318 29566 29566 I BBS_IntentReceiver: Received action(android.intent.action.DROPBOX_ENTRY_ADDED)
11-14 08:40:19.318 29566 29566 I BBS_IntentReceiver: Start BBoxsBoxService
11-14 08:40:19.333 29591 29591 E STBMonitorReceiver: dropboxTag:data_app_crash;dropboxTime:1573717219312
11-14 08:40:19.347 29566 29727 I android_os_HwBinder: HwBinder: Starting thread pool for default::vendor.fih.hardware.box@1.0::IBox
```

**Comment by freeplant** [comment]
We will check the problem.

**Comment by freeplant** [comment]
It is fixed in 2.2.21.
